code/modules/fish/fish_items.dm

```diff
@@ -159,8 +159,9 @@
 
 /obj/item/shard/shark_teeth/Initialize(mapload)
```

Comment:
I think it would be better to have a proc called in `/obj/item/shard` which sets the initial `icon_state`, this you will then override in the `shark_teeth` instance to just set it to `"teeth"`. Using `update_icon_state` is possible, but this means that calling that on `/obj/item/shard` will randomize the icon for normal shards.

Reply:
Don't think i got it, how would that work?

Reply:
Something like this (pseudo code)
```dm
/shard/Initialize()
 ...
 set_initial_icon_state()
 ...

/shard/proc/set_initial_icon_state()
 icon_state = pick("large", "medium", "small")

/shard/shark_teeth/set_initial_icon_state()
 icon_state = "teeth"
```
`set_initial_icon_state` is here the proc that you'll override to set the `icon_state` to something different. `set_initial_icon_state` can be named something else as long as it is clear what it does.
You can also choose to include the `pixel_x` and `pixel_y` code in there if you feel like it belongs together.

Reply:
Like this?